src/button/button-group.scss

```diff
@@ -9,25 +9,21 @@
 
   $childSelector: 'input, button'; // target nested inputs and buttons
 
-  > * {
+  .iui-button-group-item {
     display: flex;
+    position: relative;
 
-    #{$childSelector} {
-      position: relative;
-
-      &:hover {
-        z-index: 1;
-      }
-
-      &:focus {
-        z-index: 2;
+    &:hover {
+      z-index: 1;
+    }
 
-        &:where(:not(:focus-visible):not(:hover)) {
-          z-index: revert;
-        }
-      }
+    &:focus-within {
+      z-index: 2;
+    }
 
-      &:disabled {
+    > * {
+      &:disabled,
```

Comment:
So now hover is broken.
![image](https://user-images.githubusercontent.com/81580355/151154383-edf7dfeb-ed3a-42ad-a609-1934ac57c490.png)

![image](https://user-images.githubusercontent.com/81580355/151154409-6c315dfe-9de3-4c8f-a262-5c40d0b778ff.png)

Reply:
Should be fixed now, I moved it to the same level as `disabled`